src/Yaapii.Atoms/IO/Directory.cs

```diff
@@ -97,13 +144,26 @@ public DirectoryOf(IScalar<string> path)
         /// <returns></returns>
         public IEnumerator<string> GetEnumerator()
         {
-            return
+            return new Ternary<bool, IEnumerator<string>>(
+                this._recursive,
+                new Sorted<string>(
+                    new Joined<string>(
+                        Directory.EnumerateFiles(_dir.Value()),
+                        new Joined<string>(
+                            new Mapped<string, IEnumerable<string>>(
+                                (subDirectory) => new DirectoryOf(subDirectory, true),
+                                Directory.EnumerateDirectories(_dir.Value())
```

Comment:
A simpler approach is 
Directory.EnumerateFiles(
    path, "*.*", SearchOption.AllDirectories)